### PR TITLE
725 procurement shop

### PR DESCRIPTION
**Pull request by xlorepdarkhelm**
## What changed

Added Procurement Shops to the backend, with new endpoints. Also added in functionality for dataclass-based models.

## Issue

[Implement (static) "Procurement Shop" data structure and data (model, endpoint, etc)#725](https://app.zenhub.com/workspaces/opre-ops-62d197b4c468970013bcd2ec/issues/gh/hhs/opre-ops/725)

## How to test

The endpoints should be set up correctly by hooking up to postman or swagger or whatever.

## Screenshots

n/a

## Links

n/a

### CODE REVIEW

File: backend/data_tools/Pipfile [hunk 1]
```diff
@@ -12,6 +12,7 @@ cfenv = "==0.5.3"
 sqlalchemy-mixins = "==1.5.3"
 typing-extensions = "==4.4.0"
 pytest-docker = {extras = ["docker-compose-v2"], version = "*"}
+desert = "*"
```
Comment (tdonaworth) [inline backend/data_tools/Pipfile:15]: Need to pin the version

File: backend/ops_api/.flake8 [hunk 2]
```diff
@@ -1,6 +1,6 @@
 [flake8]
 select = B,B9,BLK,C,E,F,S,W
-ignore = E203,E501,W503
+ignore = E203,E501,W503,BLK100
```
Comment (tdonaworth) [inline backend/ops_api/.flake8:3]: Not sure we want to ignore Black formatting changes

**Review by tdonaworth — CHANGES_REQUESTED** [review]

File: backend/ops_api/.flake8 (hunk 2 above)
Comment (xlorepdarkhelm) [reply backend/ops_api/.flake8]: fixed.

File: backend/data_tools/Pipfile (hunk 1 above)
Comment (xlorepdarkhelm) [reply backend/data_tools/Pipfile]: fixed.

**Review by tdonaworth — APPROVED** [review]

**Pull request merged**